apps/mobile/src/state/chat/utils/addRealLifeInfoToChat.ts

```diff
@@ -27,19 +27,21 @@ export default function addRealLifeInfoToChat(
   contactRevealData: ContactReveal | undefined
 ): (chat: ChatWithMessages) => ChatWithMessages {
   return (chat) => {
-    const dataFromIdentityReveal =
-      processTradeChecklistIdentityRevealMessageIfAny(identityRevealData)
+    if (identityRevealData) {
+      const dataFromIdentityReveal =
+        processTradeChecklistIdentityRevealMessageIfAny(identityRevealData)
 
-    const dataFromContactReveal =
-      processTradeChecklistContactRevealMessageIfAny(
-        contactRevealData,
-        chat.chat.otherSide.realLifeInfo
-      )
-
-    if (dataFromIdentityReveal)
       return setRealLifeInfo(dataFromIdentityReveal)(chat)
-    if (dataFromContactReveal)
+    }
+
+    if (contactRevealData) {
+      const dataFromContactReveal =
+        processTradeChecklistContactRevealMessageIfAny(
+          contactRevealData,
+          chat.chat.otherSide.realLifeInfo
+        )
       return setRealLifeInfo(dataFromContactReveal)(chat)
+    }
```

Comment:
What if identityRevealData and contactRevealData are both nonnull? Only identityRevealData will be processed in that case

Reply:
Can we split it into two? Otherwise I guaratee you, in the future, one of us will call it, thinking it does both.